lib/autokey/service.py

```diff
@@ -489,28 +497,37 @@ def execute(self, script: model.Script, buffer=''):
         if script.path is not None:
             # Overwrite __file__ to contain the path to the user script instead of the path to this service.py file.
             scope["__file__"] = script.path
-        try:
-            exec(script.code, scope)
-        except Exception as e:
-            logger.exception("Script error")
-            self.error = "Script name: '{}'\n{}".format(script.description, traceback.format_exc())
-            self.app.notify_error("The script '{}' encountered an error".format(script.description))
+        self._execute(scope, script)
 
         self.mediator.send_string(trigger_character)
 
     @threaded
-    def execute_path(self, path: pathlib.Path, buffer=''):
+    def execute_path(self, path: pathlib.Path):
         logger.debug("Script runner executing: {}".format(path))
         scope = self.scope.copy()
-        # scope["store"] = script.store
         # Overwrite __file__ to contain the path to the user script instead of the path to this service.py file.
         scope["__file__"] = path.resolve()
+        self._execute(scope, path)
+
+    def _record_error(self, error: model.ScriptErrorRecord):
+        self.error_records.append(error)
+        self.app.notify_error(error)
+
+    def _execute(self, scope, script: typing.Union[model.Script, pathlib.Path]):
+        script_code = script.read_text() if isinstance(script, pathlib.Path) else script.code
```

Comment:
This is a nice refactor of the two exec methods.

Reply:
Would be easier to read as `script_code = self._get_script_code(script)` (new func)

Reply:
I’m quick to write ternary operators…
And putting my one-liner into a separate function feels odd.

How about this?
```python
if isinstance(script, pathlib.Path):
    script_code = script.read_text()
else:
    script_code = script.code
```
or even 
```python
if isinstance(script, pathlib.Path):
    script_code = script.read_text()
elif isinstance(script, model.Script):
    script_code = script.code
else:
    raise TypeError(
        "Unknown script type passed in, expected one of "
        "[autokey.model.Script, pathlib.Path], got {}".format(
              type(script))
    )
```
Especially the second one warrants a separate function.

Reply:
Seems we may as well do the second one.

I can't stand ternaries except in python, but when the clauses are long it hinders readability